### ISSUE TITLE
Multilang system in web

### ISSUE THREAD

**Issue by CrazyHackGUT** [SB-MaterialAdmin/Web#111]
Make a multilang system in web.
What we need?

- Replace all substring in templates, codes to calling functions
- Write a class **OR** download ready class
- Translate to english.

**Comment by matheussaliba** [comment]
Hi, if you want (i think you do) i can help you with english translation, but since i'm a little rusty with github, what do i have to actually do? clone the files translate them and upload them through here?

**Comment by CrazyHackGUT** [comment]
After prepared russian translation, just download russian locale, translate on English, and send me.

**Comment by tibehrius** [comment]
Matheus is our official translator ;D
We look forward to getting the translations.

Thanks man!

**Comment by Moshikot3** [comment]
I'm currently working on it.
https://puu.sh/ywcev/74d973acb2.png

**Comment by CrazyHackGUT** [comment]
@Moshikot3 Very nice, but please, add old links in footer. This is not my demand. This is SB License demand.

**Comment by BlackYuzia** [comment]
@CrazyHackGUT Or donate 10$ and you can delete this shit 🌚

**Comment by Moshikot3** [comment]
@CrazyHackGUT i'll add it when i'll finish translating, and i'll also add a credit for me 🥇

**Comment by tibehrius** [comment]
Hello guys.

How is the progress of the task? @CrazyHackGUT , do you need any help? We are looking forward to the multi language system.

Happy New Year to everybody! ;D

**Comment by Moshikot3** [comment]
@tibehrius still working on it...
https://puu.sh/yTthc/e54173c652.png
![e54173c652](https://user-images.githubusercontent.com/8211790/34531785-27b4f3f8-f0bc-11e7-9544-751d0e4a51f0.png)

**Comment by CrazyHackGUT** [comment]
@tibehrius i prepared all functions for translating SB. It remains only to get all the texts from the code, and to move in the localization files.

**Comment by CrazyHackGUT** [comment]
Init commit: https://github.com/SB-MaterialAdmin/Web/commit/6abb29a6649f8c498574e05941a05ec3869c7340

**Comment by tibehrius** [comment]
@Moshikot3 , I also did a manual translation of the original files, but this system is not good, because whenever there is an update, it is necessary to translate the files again. That's why multi-language system is necessary.

https://prnt.sc/hy06q0

@CrazyHackGUT , I'm sure a lot of people are looking forward to this update. Thank you so much for the great work.

**Comment by CrazyHackGUT** [comment]
If someone has translated web on english, german or any other language, just send me in PM in [Telegram](https://t.me/CrazyHackGUT) files with web. I'll integrate language in default SB package.

**P.S.**: Do not forget to specify when sending archive, which language you have translated the web panel.

**Comment by tibehrius** [comment]
Any predictions for the release of the multi-language version?

**Comment by CrazyHackGUT** [comment]
I'm busy now. Studying... Please, wait...

**Comment by tibehrius** [comment]
Sure. Tks man ;D

**Comment by tallanvor** [comment]
I know this feel, bro....

2018-02-02 21:22 GMT+02:00 tibehrius <notifications@github.com>:

**Comment by NWayneHD** [comment]
Guys if somebody has translated to English or German or Spanish I can translate it into Polish. So please send me if you could.

**Comment by tibehrius** [comment]
Did you give up the translation project? :(

**Comment by CrazyHackGUT** [comment]
@tibehrius 
Yeah, we continue work after implementing new templator (see #161) and replace XAJAX.

**Comment by Moshikot3** [comment]
I already have a translated version of this sourcebans, contact me if you need any help.

**Comment by fzr87** [comment]
> I already have a translated version of this sourcebans, contact me if you need any help.

I want english version :)

**Comment by tibehrius** [comment]
@CrazyHackGUT Can I help in any way? I'm so looking forward to the multi language system :(

Please let me know if there is any way to help, and I will promptly work for it.

Best regards.

**Comment by genuine96** [comment]
anything  new? i also need english version of web. i actually translated everything (php + tpl), but sb_callback is acting weird. i am sure that i translated everything on it, but some callbacks still shows on russian

**Comment by Moshikot3** [comment]
> anything new? i also need english version of web. i actually translated everything (php + tpl), but sb_callback is acting weird. i am sure that i translated everything on it, but some callbacks still shows on russian

I have SB_Materials 90% translated, but the translated version is from 2018.. you can contact my mail at: **moshikot015@gmail.com**

I can give it to you free of charge because I have no use of this anymore...

**Comment by genuine96** [comment]
> > anything new? i also need english version of web. i actually translated everything (php + tpl), but sb_callback is acting weird. i am sure that i translated everything on it, but some callbacks still shows on russian
> 
> I have SB_Materials 90% translated, but the translated version is from 2018.. you can contact my mail at: **[moshikot015@gmail.com](mailto:moshikot015@gmail.com)**
> 
> I can give it to you free of charge because I have no use of this anymore...

sent email
